src/ring_jwt_middleware/core.clj

```diff
@@ -74,84 +100,100 @@
           (when (pos? nb-ms)
             (str nb-ms "ms")))
          (remove nil?)
-         (str/join " "))))
+         (string/join " "))))
+
+(s/defn current-epoch! :- s/Num
```

Comment:
Did you consider threading an optional `current-epoch!` argument to the middleware to override using lexical scope? Perhaps we can provide an alternative stubbing API from `with-redefs`.